From 6053155d19335e5a501d186fde4bd24eaf93724e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kacper=20Michaj=C5=82ow?= <kasper93@gmail.com>
Date: Tue, 29 Sep 2026 11:34:15 +0200
Subject: [PATCH] avformat/hls: use the playlist of the new variant in
 ensure_playlist()
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

new_playlist() returns the known playlist for a duplicate URL, which is not
necessarily the last one in c->playlists. Take it from the variant, so media
tags of a Master Playlist do not end up in an unrelated playlist.

Fixes: 54f0296dfe
Signed-off-by: Kacper Michajłow <kasper93@gmail.com>
---
 libavformat/hls.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/libavformat/hls.c b/libavformat/hls.c
index 730e8b98fefb..97755da3a3ed 100644
--- a/libavformat/hls.c
+++ b/libavformat/hls.c
@@ -657,9 +657,10 @@ static int ensure_playlist(HLSContext *c, struct playlist **pls, const char *url
 {
     if (*pls)
         return 0;
-    if (!new_variant(c, NULL, url, NULL))
+    struct variant *var = new_variant(c, NULL, url, NULL);
+    if (!var)
         return AVERROR(ENOMEM);
-    *pls = c->playlists[c->n_playlists - 1];
+    *pls = var->playlists[0];
     return 0;
 }
 
